backend/users/views.py

```diff
@@ -243,3 +244,98 @@ def change_password_view(request):
         return render(request, "users/change_password.html", {"error": error})
 
     return render(request, "users/change_password.html")
+
+
+##############
+##### Friends Methods
+##############
+
+
+@jwt_login_required
+def add_friend(request, username):
+    friend = get_object_or_404(CustomUser, username=username)
+    request.user.add_friend(friend)
```

Comment:
what happens if this request fails, like if the friend does not exist?

Reply:
![image](https://github.com/user-attachments/assets/3242f638-4580-49f0-b96d-40317824fc3e)

Reply:
this is in Debug mode true; if it false it should go to 404 page.

Reply:
i'll find a way for the user to get notified in the page itself instead